### ISSUE TITLE
WRITE_BIGDECIMAL_AS_PLAIN setting ignored in play.libs.Json despite enabling on the ObjectMapper provided. Play 2.4.0

### ISSUE THREAD

**Issue by ghost** [playframework/playframework#4763]
I originally filed issue https://github.com/FasterXML/jackson-databind/issues/842 with the Jackson project. However it appears this is a bug in how the Play Framework is serializing values. Please see the Jackson issue for more context.

If I execute the following:

```java
ObjectMapper mapper = new ObjectMapper().enable(SerializationFeature.WRITE_BIGDECIMAL_AS_PLAIN);
Json.setObjectMapper(mapper);
Map<String, BigDecimal> m = new HashMap<>(); 
m.put("test", BigDecimal.valueOf(0.0000000005)); 
System.out.println(Json.stringify(Json.toJson(m)));
```

The output is:

```json
{"test":5E-10}
```

However with the WRITE_BIGDECIMAL_AS_PLAIN feature enabled, it should be calling the ```BigDecimal.toPlainString()``` method which would not print it in scientific notation. So the expected output is: 

```json
{"test":0.0000000005}
```

**Comment by cchantep** [comment]
Weird ... it should be ok according https://github.com/playframework/playframework/pull/3784

**Comment by cchantep** [comment]
Jackson `ObjectMapper` is not honoring the feature on `.valueToTree`.

```scala
    public static JsonNode toJson(final Object data) {
        try {
            return mapper().valueToTree(data);
        } catch(Exception e) {
            throw new RuntimeException(e);
        }
    }
```

**Comment by gmethvin** [comment]
This appears to be fixed in Play 2.5. I'm not actually sure if it was an issue with Jackson or something else, but running that code above yields the expected result now.
